jfix-stdlib-concurrency/src/main/java/ru/fix/stdlib/concurrency/threads/ReschedulableScheduler.java

```diff
@@ -46,17 +46,16 @@ private void detachIndicators() {
     }
 
     /**
-     * change execution by schedule type
+     * Same as {@link #schedule(DynamicProperty, DynamicProperty, Runnable)}
+     * but doesn't throw exception if {@link ReschedulableScheduler#shutdown()} was executed
      *
-     * @return result task from executionService
+     * @return result task from executionService or null if {@link ReschedulableScheduler#shutdown()} was executed
      */
-    public ScheduledFuture<?> schedule(DynamicProperty<Schedule> scheduleSupplier,
-                                       DynamicProperty<Long> startDelay,
-                                       Runnable task) {
-
+    public ScheduledFuture<?> scheduleIfNotShutdown(DynamicProperty<Schedule> scheduleSupplier,
```

Comment:
scheduleIfNotShutdown looks like it is a thread safe atomic method, but actually it is not. 
Seems like you can invoke shutdown() and scheduleIfNotShutdown() from different thread concurrently and end up with inconsistent result. 
Idea to enrich scheduler with atomic semantic of sheduleIfNotShutdown/shutdown method is good. But current if check on volatile isShutdown is not enough.  A am afraid that we would have to use blocking synchronized or lock to implement such semantic.

Reply:
Yes, it looks like we need lock. I used ReadWriteLock, because it's important to make possible parallel launching tasks.